src/Umbraco.Web.UI.Client/src/common/services/blockeditormodelobject.service.js

```diff
@@ -447,7 +447,7 @@
                     if (scaffold) {
                         blocks.push({
                             blockConfigModel: blockConfiguration,
-                            elementTypeModel: scaffold.documentType
+                            elementTypeModel: {icon: scaffold.icon, name: scaffold.contentTypeName, description: scaffold.contentTypeDescription}
```

Comment:
So was scaffold.documentType null here? Where is scaffold.documentType assigned? I feel like all of this can be resolved without any of these changes by using the EntityController (i.e. entityResource) since you should be able to get this basic information about a Document Type there even without access to Settings. That is how most property editors work.

Reply:
Yes scaffold.documentType was null from the request made to `/umbraco/backoffice/UmbracoApi/Content/GetEmptyByKey?contentTypeKey=XXX` but the top level properties were mostly there with the exception of the datatype description hence I added it.